core/src/kiso.rs

```diff
@@ -0,0 +1,143 @@
+            receiver,
+            state,
+            logger,
+        };
+        tokio::spawn(async move { actor.run().await });
+
+        Self { sender }
+    }
+
+    /// Fetch the [`ConfigurationDTO`] from the actor's state.
+    ///
+    /// # Errors
+    /// If communication with actor fails.
+    pub async fn get_dto(&self) -> Result<ConfigurationDTO, Error> {
+        let (send, recv) = oneshot::channel();
+        let msg = Message::GetDTO { respond_to: send };
+        let _ = self.sender.send(msg).await;
+        let dto = recv.await?;
+        Ok(dto)
+    }
+
+    /// Update the configuration state, applying side effects.
+    ///
+    /// Awaits until the update is applied.
+    ///
+    /// # Errors
+    /// - If updating failure occurs
+    /// - If communication with actor is failed
+    pub async fn update_with_dto(&self, dto: ConfigurationDTO) -> Result<(), Error> {
+        let (send, recv) = oneshot::channel();
+        let msg = Message::UpdateWithDTO {
+            dto,
+            respond_to: send,
+        };
+        let _ = self.sender.send(msg).await;
+        recv.await?
+    }
+}
+
+enum Message {
+    GetDTO {
+        respond_to: oneshot::Sender<ConfigurationDTO>,
+    },
+    UpdateWithDTO {
+        dto: ConfigurationDTO,
+        respond_to: oneshot::Sender<Result<(), Error>>,
+    },
+}
+
+/// TODO
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    /// TODO
+    #[error("cannot dynamically update the configuration")]
+    Update(#[from] iroha_logger::ReloadError),
+    /// TODO
+    #[error("cannot get actor's response")]
+    Communication(#[from] oneshot::error::RecvError),
+}
+
+impl From<LoggerError> for Error {
+    fn from(value: LoggerError) -> Self {
+        match value {
+            LoggerError::LevelReload(err) => Self::from(err),
+            LoggerError::Communication(err) => Self::from(err),
+        }
+    }
+}
+
+struct Actor {
+    receiver: mpsc::Receiver<Message>,
+    state: Configuration,
+    logger: LoggerHandle,
+}
+
+impl Actor {
+    async fn run(&mut self) {
+        while let Some(msg) = self.receiver.recv().await {
+            self.handle_message(msg).await
+        }
+    }
+
+    async fn handle_message(&mut self, msg: Message) {
+        match msg {
+            Message::GetDTO { respond_to } => {
+                let dto = ConfigurationDTO::from(&self.state);
+                let _ = respond_to.send(dto);
+            }
+            Message::UpdateWithDTO {
+                dto:
+                    ConfigurationDTO {
+                        logger: LoggerDTO { level: new_level },
+                    },
+                respond_to,
+            } => {
+                if let Err(err) = self.logger.reload_level(new_level).await {
+                    let _ = respond_to.send(Err(err.into()));
+                    return;
+                }
+                self.state.logger.level = new_level;
+                let _ = respond_to.send(Ok(()));
```

Comment:
So, you mean to use fire-and-forget configuration updates propagation.

Currently I designed `Kiso` in kind of transactional (but not atomic) way: it first tries to apply all updates, and only then updates the actual state. If something goes wrong during the update, it will return an error. This way, if you `POST /configuration`, `200 OK` will mean that the changes are applied, and if not, an error will return.

With fire-and-forget, `POST /configuration` will not return an error, but will always return `202 ACCEPTED`, and `GET /configuration` will return updated config which might not be really applied in the system.

I am fine either way and not sure which option is better. IMO the current design makes Iroha more reliable.

Reply:
Looks like this error pop up when either logger layer was either poisoned or dropped, not smt we can handle meaningfully.